### ISSUE TITLE
Training dataset identification for sct deep learning based methods

### ISSUE THREAD

**Issue by NathanMolinier** [spinalcordtoolbox/spinalcordtoolbox#4046]
### Backgroung

SCT provides several functions based on deep learning methods : `sct_deepseg_sc`, `sct_label_vertebrae`... 
The performances of these methods mainly depend on the quality and the quantity of the data used for the training of the networks.

### Issue

Currently there is no simple way to find out which datasets were used for the training of each deep learning based method.

### Use examples

This information could be relevant:
- to compare with more accuracy `sct_functions` with other deep learning based methods: here is an [example](https://github.com/spinalcordtoolbox/spinalcordtoolbox/issues/3793)
- to retrain `sct_functions` with larger and more complex datasets to improve their performances
- to train new models with the same type of training: preprocessing steps and datasets used

**Comment by joshuacwnewton** [comment]
> Currently there is no simple way to find out which datasets were used for the training of each deep learning based method.

I agree that it would be beneficial to document the methods used for each of our (older) deep learning models. (Ideally in the argparse help for each CLI script?)

Though, I will say, most of SCT's methods have associated papers with them, where the dataset/training methods are detailed. The best page to look for associated papers is: https://spinalcordtoolbox.com/user_section/citing_sct.html#command-line-tools

**Comment by joshuacwnewton** [comment]
Small note: This issue partially overlaps with:

- https://github.com/spinalcordtoolbox/spinalcordtoolbox/issues/1767

**Comment by jcohenadad** [comment]
the info is not in the paper, unfortunately-- we need to dig in:
- our duke folders (student's folder, projects folder, etc.)
- our many repos

**Comment by joshuacwnewton** [comment]
> the info is not in the paper, unfortunately

Thanks for clarifying! That's a shame... :sweat: 

I can create a list of each deep learning-based CLI script, along with their associated repos and papers, to see what information is missing for each script.

**Comment by jcohenadad** [comment]
> I can create a list of each deep learning-based CLI script, along with their associated repos and papers, to see what information is missing for each script.

i think it's essentially deepseg_sc and deepseg_gm, which will eventually be replaced with newer models, and i hope that our practices for reproducibility have improved over the years, so data will be more tracable in the future. So i'm not sure it is worth doing an exhaustive listing-- for now, let's focus on deepseg_sc and label_vertebrae with hourglass (trained by Reza). @NathanMolinier I _think_ the hourglass stuff was trained using the spine-generic T1w and T2w

**Comment by NathanMolinier** [comment]
I will have to check for sct_deepseg_sc and sct_label_vertebrae to be fair in the comparison of the methods.
